### ISSUE TITLE
Registry not generated on installation

### ISSUE THREAD

**Issue by reset** [hexpm/hex#15]
After installing hex with `mix local.hex --force` on my build server I noticed that other basic mix commands began to fail due to an ENOENT error when attempting to read the Hex registry (`~/.mix/hex.ets`).

I alleviated this postfixing a build step to my build jobs that checks for the existence of the hex registry and generates it if it doesn't exist by running `mix hex.search {any_package}`.

It appears that either the registry should be generated on installation or that it shouldn't be loaded if a task is being run which doesn't require it.

**Comment by dvmason** [comment]
`mix local.hex
mix hex.info`

Seems to clear up registry problems similar to above.  The second line is needed, too

**Comment by Gyllsdorff** [comment]
I just had the same problem on a new Windows 10 laptop.
 
```
C:\workspace\[PROJECT_NAME]>mix compile
Could not find Hex, which is needed to build dependency :erlsom
Shall I install Hex? [Yn] Y
* creating c:/Users/matti/.mix/archives/hex-0.11.1.ez
** (Mix) Failed to open Hex registry ({:read_error, {:file_error, 'c:/Users/matti/.hex/registry.ets', :enoent}})

C:\workspace\[PROJECT_NAME]>mix hex.info
Hex:    0.11.1
Elixir: 1.2.3
OTP:    18.2.1

Built with: Elixir 1.2.3 and OTP 18.2.4

Registry file available (last updated: 2016-03-11 10:17:13)
Size: 1636kB (compressed 531kb)
Packages #: 1688
Versions #: 7577

C:\workspace\[PROJECT_NAME]>mix compile

C:\workspace\[PROJECT_NAME]>
```

I installed Elixir and Erlang using the [websetup](https://s3.amazonaws.com/s3.hex.pm/elixir-websetup.exe). The system is a new windows 10 Dell laptop without any modifications. All commands was entered in a Admin command prompt.

**Comment by ericmj** [comment]
@Gyllsdorff Did you run `mix compile` before fetching dependencies?

**Comment by Gyllsdorff** [comment]
Yes, I did. Running `mix deps.get` also fixes the error. 

I looked through my notes and I have encountered the `enoent` error once before. I created a demo app using `mix new example_app`, added erlsom as a dependency, ran `mix test` and got the same error. I wrote that it fixed itself suddenly, I probably ran `mix deps.get` without thinking about it. 

I can recreate it by remove `%UserProfile%\.hex` and `%UserProfile%\.mix` and creating a new project like the above.
